src/platform/posix/posix_pollq_epoll.c

```diff
@@ -134,19 +141,27 @@ nni_posix_pfd_arm(nni_posix_pfd *pfd, unsigned events)
 
 	nni_mtx_lock(&pfd->mtx);
 	if (!pfd->closing) {
-		struct epoll_event ev;
+		struct epoll_event *ev;
 		pfd->events |= events;
 		events = pfd->events;
 
-		ev.events   = events | NNI_EPOLL_FLAGS;
-		ev.data.ptr = pfd;
+		if ((ev = NNI_ALLOC_STRUCT(ev)) == NULL) {
+			return (NNG_ENOMEM);
```

Comment:
From what I can tell it is ok to introduce the `NNG_ENOMEM` return code here. Seems the top-level functions already state that `NNG_ENOMEM` can be returned from them. Not entirely confident in my quick glance over the source code though.